tangelo/linq/target/target_qulacs.py

```diff
@@ -40,56 +40,79 @@ def simulate_circuit(self, source_circuit: Circuit, return_statevector=False, in
         equivalent gates.
 
         Args:
-            source_circuit: a circuit in the abstract format to be translated
+            source_circuit (Circuit): a circuit in the abstract format to be translated
                 for the target backend.
-            return_statevector(bool): option to return the statevector as well,
+            return_statevector (bool): option to return the statevector as well,
                 if available.
-            initial_statevector(list/array) : A valid statevector in the format
+            initial_statevector (list/array) : A valid statevector in the format
                 supported by the target backend.
+            save_mid_circuit_meas (bool): Save mid-circuit measurement results to
+                self.mid_circuit_meas_freqs. All measurements will be saved to
+                self.all_frequencies, with keys of length (n_meas + n_qubits).
+                The leading n_meas values will hold the results of the MEASURE gates,
+                ordered by their appearance in the source_circuit.
+                The last n_qubits values will hold the measurements performed on
+                each of qubits at the end of the circuit.
 
         Returns:
             dict: A dictionary mapping multi-qubit states to their corresponding
                 frequency.
             numpy.array: The statevector, if available for the target backend
                 and requested by the user (if not, set to None).
         """
-
         translated_circuit = translate_c(source_circuit, "qulacs",
-            output_options={"noise_model": self._noise_model})
+                output_options={"noise_model": self._noise_model, "save_measurements": save_mid_circuit_meas})
 
         # Initialize state on GPU if available and desired. Default to CPU otherwise.
         if ('QuantumStateGpu' in dir(self.qulacs)) and (int(os.getenv("QULACS_USE_GPU", 0)) != 0):
             state = self.qulacs.QuantumStateGpu(source_circuit.width)
         else:
             state = self.qulacs.QuantumState(source_circuit.width)
+
+        python_statevector = None
```

Comment:
This become has become difficult to understand:
Can we please have a comment before each major branch that tells what it does succinctly ? Just enough to understand the difference between the main branches if the code.

Reply:
ok